src/args.rs

```diff
@@ -111,6 +111,10 @@ pub struct CliArgs {
     #[clap(short = 'u', long = "upload-files")]
     pub file_upload: bool,
 
+    /// Restrict upload directories
+    #[clap(long = "restrict-upload-dir", requires = "file-upload", value_hint = ValueHint::FilePath)]
+    pub restrict_upload_dir: Vec<PathBuf>,
```

Comment:
```suggestion
    /// Allowed upload directories
    ///
    /// If this is set, uploads are only allowed into the provided directories. 
    #[clap(long, requires = "file-upload", value_hint = ValueHint::FilePath)]
    pub allowed_upload_dir: Vec<PathBuf>,
```

Reply:
I believe this explanation and naming makes more sense. Please change it like that throughout your code unless you have a better suggestion for the naming.

Reply:
Yes, this is clearer.